### PR TITLE
MEAL-21 added stock schema

### PR DESCRIPTION
**Pull request by otuffaha**
I added the stock schema based on the document in the slack documents channel. I created a quantity class for these stock items.

### CODE REVIEW

**Comment by otuffaha** [comment]
Please do not merge this PR. I will merge it after receiving constructive feedback

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "foodStock")
+public class FoodStock {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: please check this mapping to ensure it is accurate.

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  public void setStockItems(Set<StockItem> stockItems) {
+    this.stockItems = stockItems;
+  }
+
+  /**
+   * Retrieves the quantity of the desired stock item in this food stock
+   * @param stockItem the desired stock item
+   * @return Optional of the quantity for the desired stock item
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: is this the desired functionality we want?

File: src/main/java/com/app/eece4792mealmaster/models/Quantity.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package com.app.eece4792mealmaster.models;
+
+import java.util.Objects;
+
+/**
+ * Represents a quantity of something (in particular, a quantity of a stock item)
+ */
+public class Quantity {
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/models/Quantity.java]: does a quantity need anything else?

File: src/main/java/com/app/eece4792mealmaster/models/StockItem.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+import javax.persistence.MapsId;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "stockItem")
+public class StockItem {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/models/StockItem.java]: please check this mapping for me

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.Set;
+
+import java.util.stream.Collectors;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: am i missing a mapping/field that associates food stocks to generic foods?

Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: naming conventions for tableNames is to use underscore e.g. food_stock

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java (hunk 1 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: Need a mapping to user

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java (hunk 2 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: Potentially use "@transient" tag here. idk about .stream() but it should be as simple as folding over this.stockItems and accumulating the quantities there

File: src/main/java/com/app/eece4792mealmaster/models/Quantity.java (hunk 3 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/models/Quantity.java]: Units probably aren't necessary.  We'll go with quantity as a double representing the number of servings of the generic food. We can discuss how we want to store the actual data, but regardless units should not be handled here

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java (hunk 5 above)
Comment (otuffaha) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: done

File: src/main/java/com/app/eece4792mealmaster/models/Quantity.java (hunk 3 above)
Comment (otuffaha) [reply src/main/java/com/app/eece4792mealmaster/models/Quantity.java]: done

File: src/main/java/com/app/eece4792mealmaster/models/FoodStock.java (hunk 1 above)
Comment (otuffaha) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: please check the mapping

Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: I don't see the user mapping

Comment (otuffaha) [reply src/main/java/com/app/eece4792mealmaster/models/FoodStock.java]: i withheld the push, check again later tonight

**Comment by otuffaha** [comment]
This PR is part of MEAL-33's PR: https://github.com/Team-W4/eece4792-meal-master/pull/7

**Pull request closed without merging**